core/state/state_object.go

```diff
@@ -148,7 +148,7 @@ func (s *stateObject) touch() {
 	s.db.journal.append(touchChange{
 		account: &s.address,
 	})
-	if s.address == ripemd {
+	if bytes.Equal(s.address.Bytes(), ripemd.Bytes()) {
```

Comment:
don't we have a new address Equals function? We should use that instead of comparing bytes like this

Reply:
s.address is an `InternalAddress` which does not have an `Equals` function. I think `bytes.Equal` makes more sense here but I could be convinced otherwise

Reply:
i guess it depends on if we would have a situation where we need to compare the InternalAddress. Then it would be nice to use Equals.

Reply:
InternalAddress can be compared with InternalAddress with the == operator because they are arrays

Reply:
okay, thanks